### DISCUSSION TITLE
How should we implement the target contract to make it possible to batch-create streams without reverting the entire tx if a subset of the streams fail to be created?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/v2-periphery#6] [General]
We will write [PRBProxy](https://github.com/paulrberg/prb-proxy)-powered target contracts to batch-create streams.

The question is: how to implement the batch-create streaming function without reverting the entire tx if a subset of the streams fail to be created?

**Comment by PaulRBerg** [comment]
I think the solution for this is [try/catch](https://ethereum.stackexchange.com/q/78562/24693).

**Comment by PaulRBerg** [comment]
@andreivladbrg can you post an update here? I recall that we've talked about this at one point and you ended up reverting the whole tx even if just one stream failed.

**Reply by andreivladbrg** [reply]
Initially I implemented the create multiple functions with a try/catch statement but I realised that this is wrong because it will lead us into a bug. In the current version of the functions we are transferring a `totalAmount`(an amount for all the streams) in the periphery contract and if one of the stream creation would fail, money would be lost because the function would not revert.

**Reply by PaulRBerg** [reply]
Well, it doesn't _necessarily_ lead to a bug. It just leads to a situation that would have to be handled explicitly to transfer the funds back to the user.

However, in the context of PRBProxy, this wouldn't even lead to a problem, because any funds in the PRBProxy can always be transferred back to the user (with a bespoke target contract).

Whatever the case - I am happy to keep the behavior as you implemented it, @andreivladbrg (revert even if just one stream fails).

**Comment by PaulRBerg** [comment]
Locking as we have decided to revert the entire transaction even if just one stream creation tx fails.
